src/main/rascal/refactor/Rename.rsc

```diff
@@ -277,12 +305,14 @@ default tuple[set[loc] defFiles, set[loc] useFiles] findOccurrenceFiles(set[Defi
 
 default set[Define] findAdditionalDefinitions(set[Define] cursorDefs, Tree tr, TModel tm) = {};
 
-default void renameDefinition(Define d, str newName, TModel tm, Renamer r) {
-    r.textEdit(replace(d.defined, newName));
+default void renameDefinition(Define d, loc nameLoc, str newName, Tree _, TModel tm, Renamer r) {
+    r.textEdit(replace(nameLoc, newName));
 }
 
-default void renameUses(set[Define] defs, str newName, TModel tm, Renamer r) {
-    for (loc u <- invert(tm.useDef)[defs.defined]) {
+default void renameUses(set[Define] defs, str newName, Tree _, TModel tm, Renamer r) {
+    for (loc u <- invert(tm.useDef)[defs.defined] - defs.defined) {
         r.textEdit(replace(u, newName));
     }
-}
\ No newline at end of file
+}
+
+default Maybe[loc] nameLocation(Tree t, Define _) = nothing();
```

Comment:
I think it should return a Maybe, but always a location. And the default should be `t@src`

Reply:
In that case if someone doesn't implement `nameLocation` for all possible definitions, whole definitions might be replaced by a single name, thus likely introducing a parse error. Is that really what we want?

Reply:
it'll be wrong in that case either way right?

Reply:
With `Maybe[loc]`, we can detect if any definition has a `nothing` name and inform the user by throwing an error, which would make the renaming invalid, but leave the program intact. If we just replace the whole definition with a name, it would be applied and result in a broken file/parse error.